app/service/rest.py

```diff
@@ -466,16 +467,23 @@ def add_user_to_service(service_id, user_id):
     user = get_user_by_id(user_id=user_id)
 
     if user in service.users:
-        error = "User id: {} already part of service id: {}".format(user_id, service_id)
-        raise InvalidRequest(error, status_code=400)
+        message = "User id: {} already part of service id: {}".format(user_id, service_id)
+        current_app.logger.info(message)
+        raise UserAlreadyInServiceError(status_code=409, message=message)
 
     data = request.get_json()
     validate(data, post_set_permissions_schema)
 
     permissions = [Permission(service_id=service_id, user_id=user_id, permission=p["permission"]) for p in data["permissions"]]
     folder_permissions = data.get("folder_permissions", [])
 
-    dao_add_user_to_service(service, user, permissions, folder_permissions)
+    try:
+        dao_add_user_to_service(service, user, permissions, folder_permissions)
+    except UniqueViolation:
+        message = f"UniqueViolation: User id: {user_id} already part of service id: {service_id}"
+        current_app.logger.info(message)
+        raise UserAlreadyInServiceError(status_code=409, message=message)
```

Comment:
I like business specific exception to raise up to business code higher in the stack, good job!

Reply:
I gave that a try but it resulted in a circular import error. So I went back to the easy solution. But thanks for the suggestion, I agree that would be best practice.